docs/time_alignment.rst

```diff
@@ -0,0 +1,77 @@
+
+The LSL stream will contain gaze data a resolution of 66 Hz.
+You can get a higher sampling rate when you're using the gaze data downloaded from
+Pupil Cloud. In order to do this, you'll need to align the timestamps collected during
+the lsl streaming to the timestamps recorded in cloud.
+
+Setup
+=====
+#. Start the lsl relay of your Pupil Invisible glasses.
+
+#. In your lsl recording software, select both the gaze data stream and the event stream.
+
+#. Start the lsl recording through your software of choice (e.g. LabRecorder).
+
+#. In your Pupil Invisible Companion App, tap the red "record" button and make sure the recording is running.
+
+#. Run your experiment/data collection.
+
+#. Stop the recording in the Pupil Invisible Companion App.
+
+#. Stop the lsl recording.
+
+#. Wait till the Gaze Data was uploaded to Pupil Cloud and the 200 Hz Gaze-Data was computed.
+
+#. Export the Gaze Data with the Raw Data Exporter Enrichment.
+
+#. You will end up with a n xdf file, containing all data recorded through lsl (including gaze and event data),
+   and one csv file for each gaze and event data downloaded from Pupil Cloud.
+
+From here, you can perform the timestamp alignments
+
+.. important::
+   If your recording is short (less than 1 minute), you should increase the frequency at which ``lsl.timesync`` events
+   are being generated. As a rule of thumb you should aim for at least 3 events being sent throughout the recording, including
+   ``recording.begin`` and ``recording.end`` events, which are generated by starting and ending the recording in the Pupil
+   Invisible Companion App.
+
+   You can change the frequency at which ``lsl.timesync`` events are being sent by setting the ``--time_sync_interval``
+   argument.
+
+Loading event data from xdf file
+=================================
+Event streams can be identified and filtered by their name.
+
+.. literalinclude:: ../examples/load_events_from_xdf.py
+   :language: python
+   :linenos:
+
+Loading event data from cloud
+==============================
+The raw data enrichment contains a csv file with event names and timestamps.
+We can load this csv file with pandas.
+
+.. literalinclude:: ../examples/load_events_from_csv.py
+   :language: python
+   :linenos:
+
+Building a linear model to map cloud time to lsl time
+======================================================
+After extracting the event names and time stamps from both series as shown above,
+you can build a linear model to translate from one time series to the other.
+
+Once the linear model was fitted, we can apply it to map the cloud timestamps to lsl timestamps.
+
+.. hint::
+   If you want to invert the mapping, to transform lsl timestamps to cloud timestamps,
+   you'll have to change the order of arguments in ``time_mapper.fit`` to
+   ``time_mapper.fit(filtered_lsl_event_times, filtered_cloud_event_times.reshape(-1, 1))``
+
+.. literalinclude:: ../examples/linear_time_model.py
+  :language: python
+  :linenos:
```

Comment:
Given that the `linear_time_model.py` contains the content of the other two examples, it might be easier to maintain only one "reference" example and make use of [`literalinclude`'s `:lines:`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude) directive to only include the lines you need for each section.

Reply:
Thanks a lot! I was looking for that functionality but got stuck.